### PR TITLE
Local Storage

### PR DESCRIPTION
**Pull request by degisew**
### Local Storage
**In this project I did the following features:**
- When the user changes the content of any input field, the data is saved to the local storage.
- When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.
- I used a single JavaScript object to collect data from the form and to store in a local storage.

### CODE REVIEW

**Comment by degisew** [comment]
Hello reviewer, I already reported my partner's absence to the student success team, and they unpaired me. now, I am in a solo. so, please consider this while reviewing
Microverse report
 my project.

File: localStorage.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+const userEmail = document.getElementById('client-email');
+const userMessage = document.getElementById('client-message');
+
+const userInput = {
+  un: userName.value,
+  email: userEmail.value,
+  message: userMessage.value,
+};
+
+localStorage.setItem('userData', JSON.stringify(userInput));
```
Comment (Gambit142) [inline localStorage.js:11]: - [ ] The essence of the `localStorage` is to save data in it when you fill the form. A user should be able to see what they filled in the numerous form inputs on page load even if they don't submit the form. To achieve this, I would advise you to save the `userInput` object to the local storage when an input event on the form is made or when there is a change to the numerous forms. 

#### Hint: Add an input `eventlistener` to the form (preferable) or change `eventlistener` to each form inputs. After that, your callback function in the `eventlistener` should declare your `userInput` in it and save the `userInput` to the `localStorage`

File: localStorage.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+const test = localStorage.getItem('userData');
+const parsedData = JSON.parse(test);
+userName.value = parsedData.un;
+userEmail.value = parsedData.email;
+userMessage.value = parsedData.message;
```
Comment (Gambit142) [inline localStorage.js:16]: - [ ] To get data from the `localStorage`, I would advise you to add a conditional that if `parsedData` is not empty, you can assign the values of the various form inputs to the values gotten from the localStorage. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @degisew 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- Pull request summary is descriptive ✔️ 
- Form data for all inputs are stored in a single Object ✔️ 

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: localStorage.js (hunk 2 above)
Comment (degisew) [reply localStorage.js:16]: Thank you very much. you raise important thing. I will walk through it.

File: localStorage.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  return true;
+});
+
+const test = localStorage.getItem('userData');
+const parsedData = JSON.parse(test);
+if (parsedData !== null) { // to check wheather the local storage is empty or not.
+  userName.value = parsedData.un;
+  userEmail.value = parsedData.email;
+  userMessage.value = parsedData.message;
+}
```
Comment (BertrandConxy) [inline localStorage.js:23]: - [ ] Great improvement 👏. However, It looks like till now, data form data are not persisting in the local storage. I realized that the `userInput object` is being saved into local storage but its properties like un, email, and message have empty values. Take a look at the screenshot below 👇 👇 
image

I think you may consider adding the appropriate event and also check again if you are referencing well your form from the DOM. I have seen that you used `getElementByTagName` may be you may consider giving the form an ID and then reference it by its ID

**Review by BertrandConxy — CHANGES_REQUESTED** [review]
Hi @degisew , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights 💪 
- no linter errors
- correct GitHub flow is used
- good PR description


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mwanawabangona — APPROVED** [review]
Hi @degisew ,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implimenting the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**